### ISSUE TITLE
Bug: consistent-type-imports and -exports doesn't work as expected and described in docs

### ISSUE THREAD

**Issue by PatrykKuniczak** [typescript-eslint/typescript-eslint#11010]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug) and found none that matched my issue.
- [x] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Issue Description

I have problem with `inline-type-imports` for:
`@typescript-eslint/consistent-type-exports` 
and 
`@typescript-eslint/consistent-type-imports`

As you can see on my repro repo, on file `src/App.tsx` i have defined:
```ts
import {type User, type Post } from './types';
```

And on `pnpm lint` everything looks good, but it should be invalid, because `default` value for that should be:
```ts
import type { User, Post } from './types';
```

Because of default value of [Fix Style](https://typescript-eslint.io/rules/consistent-type-imports/#fixstyle)

And i'm not able to enfore that behaviour for that, even when i define:
```ts
{
 fixStyle: 'separate-type-imports'
}
```

I want to split types to other line that "normal" (const, func) imports/exports.

It only works well if i define
```ts
import { type Post } from './types';
import {User} from "./types";
```
and on `eslint --fix`
it change/move it to:
```ts
import type {User, type Post  } from './types';
```

But it also shouldn't work that way, because that `--fix` causes next error:

![Screen with error hint from WebStorm](https://github.com/user-attachments/assets/d167d7f5-d5a5-411f-9ff1-d8cae194701e)

It should change code to
```ts
import type { User, Post } from './types';
```

### Reproduction Repository Link

https://github.com/PatrykKuniczak/eslint-repro

### Repro Steps

1) clone the repo
2) `pnpm i`
3) `pnpm lint` (Is valid, but should be invalid)

### Versions

[package.json](https://github.com/PatrykKuniczak/eslint-repro/blob/master/package.json)

**Comment by bradzacher** [comment]
If you look in the docs there's a section at the end which describes related rules that will be of interest to you

https://typescript-eslint.io/rules/consistent-type-imports/#related-to

**Comment by PatrykKuniczak** [comment]
@bradzacher Thanks a lot for your help.

I was reading about conflicts between rules and etc. but i don't know why i didn't read it 😆 

`'import-x/consistent-type-specifier-style': 'error',`
helps me :)
